### PR TITLE
Finance reports

### PR DESCRIPTION
**Pull request by nottmjr24**

### CODE REVIEW

File: src/main/resources/features/theme 1/11 - finance.feature [hunk 1]
```diff
@@ -120,9 +123,77 @@ Feature: 11 - Finance
...
+          | status_description           |
+          | Successful payment received  |
+          | payment_type                 |
+    Then I record the finance report URL
+ 
+    Examples:
+      | report                | startDate | endDate | seconds  | refresh |
+      | All Payments          | 30        | 0       | 30       | 20      |
+
+
```
Comment (karlbeckettbjss) [inline src/main/resources/features/theme 1/11 - finance.feature]: please remove the extra white space at the bottom of this file

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java [hunk 2]
```diff
@@ -108,7 +108,6 @@ public WebStepDefinitions(WebDriverWrapper driverWrapper) {
 
         When("^I enter \\{([^\\}]+)\\} in the field with id \"([^\"]+)\"$", (String dataKey, String id) ->
                 driverWrapper.enterIntoFieldWithId(driverWrapper.getData(dataKey), id));
-
```
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java]: does this need to added back in?

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java [hunk 3]
```diff
@@ -203,7 +202,11 @@ public WebStepDefinitions(WebDriverWrapper driverWrapper) {
 
         And("^I delete the \"([^\"]+)\" cookie$", (String cookieName) -> driverWrapper.deleteCookie(cookieName));
 
-        And("^I wait for \"(\\d+)\" seconds$", (Integer time) -> driverWrapper.timeWait(time));
+        And("^I wait for \"([^\"]+)\" seconds$", (Integer time) -> driverWrapper.timeWait(time));
+
+        And("^I wait \"([^\"]+)\" seconds then \"([^\"]+)\" the page until \"([^\"]+)\" button displays$",
```
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:208]: New step needs to be added to the documentation

mot-automated-testsuite/docs/step-definitions.md

File: src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java [hunk 4]
```diff
@@ -1732,4 +1734,30 @@ public void switchToTab(int tabNumber) {
         ArrayList<String> tabs = new ArrayList<String>(webDriver.getWindowHandles());
         webDriver.switchTo().window(tabs.get(tabNumber));
     }
+
+    /**
+     * wait for the specified button.
+     *
+     * @param buttonText The button text
+     */
+    public void waitForButton(Integer seconds, Integer refresh, String buttonText) {
```
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java:1745]: Does this new method need a test creating for it?

mot-automated-testsuite/src/test/java/uk/gov/dvsa/mot/framework/WebDriverWrapperTest.java

File: src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java [hunk 5]
```diff
@@ -1732,4 +1734,30 @@ public void switchToTab(int tabNumber) {
         ArrayList<String> tabs = new ArrayList<String>(webDriver.getWindowHandles());
         webDriver.switchTo().window(tabs.get(tabNumber));
     }
+
+    /**
+     * wait for the specified button.
+     *
+     * @param buttonText The button text
```
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java]: missing seconds and refresh from here

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java (hunk 2 above)
Comment (nottmjr24) [reply src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java]: done

File: src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java (hunk 5 above)
Comment (nottmjr24) [reply src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java]: done

File: src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java (hunk 4 above)
Comment (nottmjr24) [reply src/main/java/uk/gov/dvsa/mot/framework/WebDriverWrapper.java:1745]: The method is called by "And I wait "<seconds>" seconds then "<refresh>" the page until "Download Report" button displays"

**Pull request merged**